PhysicsTools/UtilAlgos/src/CachingVariable.cc

```diff
@@ -53,13 +53,12 @@ VariableComputer::VariableComputer(const CachingVariable::CachingVariableFactory
 }
 
 void VariableComputer::declare(std::string var, edm::ConsumesCollector& iC) {
-  std::string aName = name_ + separator_ + var;
-  ComputedVariable* newVar = new ComputedVariable(method_, aName, arg_.iConfig, this, iC);
   if (iCompute_.find(var) != iCompute_.end()) {
     edm::LogError("VariableComputer") << "redeclaring: " << var << " skipping.";
-    delete newVar;
     return;
   }
+  std::string aName = name_ + separator_ + var;
+  ComputedVariable* newVar = new ComputedVariable(method_, aName, arg_.iConfig, this, iC);
```

Comment:
where is this variable deleted after this change?

Reply:
I think it is not false positive. If `var` is found in `iCompute` then deleting `newVar` will call destructor which will call destruct  `myComputer` at https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/UtilAlgos/interface/CachingVariable.h#L171 which I think points to `this` (https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/UtilAlgos/interface/CachingVariable.h#L161)

Reply:
> where is this variable deleted after this change?

It is not created if duplicate, and if not duplicate it's destroyed in the same place as before (I guess in `~VariableComputer`)

Reply:
@iarspider @smuzaffar please modify the PR title if it's indeed not false-positive